pkg/api/plugins.go

```diff
@@ -506,6 +506,26 @@ func (hs *HTTPServer) callPluginResource(c *models.ReqContext, pluginID, dsUID s
 	}
 	clonedReq.URL = urlPath
 
+	if dsUID != "" {
+		ds, err := hs.DataSourceCache.GetDatasourceByUID(c.Req.Context(), dsUID, c.SignedInUser, c.SkipCache)
+
+		if err != nil && !errors.Is(err, models.ErrDataSourceNotFound) {
+			handleCallResourceError(err, c)
+			return
+		}
+
+		if ds != nil && hs.DataProxy.OAuthTokenService.IsOAuthPassThruEnabled(ds) {
```

Comment:
No need to `nil` check ds here as we presume it's okay when checking for `err` above